src/app/src/main/kotlin/org/imperial/mrc/hint/db/CalibrateDataRepository.kt

```diff
@@ -0,0 +1,81 @@
+package org.imperial.mrc.hint.db
+
+import org.jooq.tools.json.JSONArray
+import org.jooq.tools.json.JSONObject
+import org.springframework.stereotype.Component
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.imperial.mrc.hint.exceptions.CalibrateDataException
+import java.sql.Connection
+import java.sql.DriverManager
+import java.sql.Statement
+import java.util.Properties
+import java.util.Locale
+import java.sql.ResultSet
+import java.sql.SQLException
+
+const val DEFAULT_QUERY = """SELECT
+age_group,area_id,
+calendar_quarter,
+indicator,
+ROUND(lower, 4) AS lower,
+ROUND(mean, 4) AS mean,
+ROUND(mode, 4) AS mode,
+sex,
+ROUND(upper, 4) AS upper
+FROM data"""
+
+interface CalibrateDataRepository
+{
+    fun getDataFromPath(path: String): JSONObject
+}
+
+@Component
+class JooqCalibrateDataRepository: CalibrateDataRepository
+{
+
+    private fun convertToJSONArray(resultSet: ResultSet): JSONArray {
+        val jsonArray = JSONArray()
+        while (resultSet.next()) {
+            val obj = JSONObject()
+            val totalRows = resultSet.metaData.columnCount
+            for (i in 0 until totalRows) {
+                obj[resultSet.metaData.getColumnLabel(i + 1)
+                    .lowercase(Locale.getDefault())] = resultSet.getObject(i + 1)
+            }
+            jsonArray.add(obj)
+        }
+        return jsonArray
+    }
+
+    private fun getDataFromQuery(conn: Connection): JSONArray {
```

Comment:
Since we always know the shape of the data we should get out of the query, I think it would be nice to define a model class e.g. `CalibrateResultRow` (see the others in the `models` folder) and have an array of those as the output here rather than just a generic `JSONArray`

Reply:
i think i have done something about this, take a look, not sure if its the right approach